lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -1120,15 +1198,31 @@ class ConvertTritonGPUOpToLLVMPatternBase {
       Location loc, ConversionPatternRewriter &rewriter, Attribute layout,
       RankedTensorType type, bool withCTAOffset) const {
     // step 1, delinearize threadId to get the base index
+    auto shape = type.getShape();
+    unsigned rank = shape.size();
     auto multiDimBase =
         emitBaseIndexForLayout(loc, rewriter, layout, type, withCTAOffset);
+    for (unsigned d = 0; d < rank; ++d) {
+      LLVM_DEBUG(DBGS() << "emitOffsetForLayout base " << d << ": ");
+      auto str = printPtrExpr(multiDimBase[d]);
+      LLVM_DEBUG(llvm::dbgs() << str << "\n");
+    }
```

Comment:
Write this as

```
LLVM_DEBUG({
  for(...) {
    blah blah;
  }
})
```

As currently written, we run printPtrExpr (and then discard the result) in non-debug mode.